src/stories/containers/Finances/components/SectionPages/ReservesWaterFallChartSection/useReservesWaterFallChart.tsx

```diff
@@ -1,22 +1,41 @@
 import { useMediaQuery } from '@mui/material';
+import { fetchAnalytics } from '@ses/containers/Finances/api/queries';
 import { nameChanged } from '@ses/containers/Finances/utils/utils';
 import { useThemeContext } from '@ses/core/context/ThemeContext';
 import lightTheme from '@ses/styles/theme/light';
 import sortBy from 'lodash/sortBy';
+import { useRouter } from 'next/router';
 import { useMemo, useState, useEffect } from 'react';
-import { builderWaterFallSeries, processDataForWaterFall } from './utils';
+import useSWRImmutable from 'swr/immutable';
+import {
+  builderWaterFallSeries,
+  getAnalyticForWaterFall,
+  processDataForWaterFall,
+  sumValuesFromMapKeys,
+} from './utils';
 
 import type { MultiSelectItem } from '@ses/components/CustomMultiSelect/CustomMultiSelect';
 import type { LegendItemsWaterFall } from '@ses/containers/Finances/utils/types';
 import type { AnalyticGranularity } from '@ses/core/models/interfaces/analytic';
 import type { Budget } from '@ses/core/models/interfaces/budget';
 
-export const useReservesWaterFallChart = (levelPath: string | null, budgets: Budget[], allBudgets: Budget[]) => {
-  const selectAll = useMemo(() => budgets.map((budget) => budget.id), [budgets]);
-
+export const useReservesWaterFallChart = (
+  levelPath: string | null,
+  budgets: Budget[],
+  allBudgets: Budget[],
+  year: string
+) => {
+  const selectAll = useMemo(() => budgets.map((budget) => budget.codePath), [budgets]);
+  const { isLight } = useThemeContext();
+  const [selectedGranularity, setSelectedGranularity] = useState<AnalyticGranularity>('monthly');
+  const isMobile = useMediaQuery(lightTheme.breakpoints.down('tablet_768'));
+  const isTable = useMediaQuery(lightTheme.breakpoints.between('tablet_768', 'desktop_1024'));
+  const router = useRouter();
+  const urlPath = Array.isArray(router.query.path) ? router.query.path.join('/') : router.query.path;
+  const codePath = urlPath ? `atlas/${urlPath}` : 'atlas';
```

Comment:
is not `codePath` and the prop `levelPath` the same thing?

Reply:
you are right